sns_ik_lib/include/sns_ik/sns_ik.hpp

```diff
@@ -78,7 +78,7 @@ namespace sns_ik {
       return m_initialized;
     }
 
-    inline bool getJointNames(std::vector<std::string> jointNames) {
+    inline bool getJointNames(std::vector<std::string>& jointNames) {
```

Comment:
While we're changing this, might as well make it `const std::vector<std::string>&`

Reply:
That would break it again because jointNames is suppose to be an output parameter and thus not constant.

If we were being strict with the coding standard, all output parameters should be pointers instead of references to make it clear to the user which are output. Making that change across the library would be a much larger change.

Reply:
Aha, fair point. I missed the fact that this is an output parameter. I agree with your pointer thought - we really should be using them here and everywhere.